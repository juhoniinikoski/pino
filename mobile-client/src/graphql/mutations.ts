import { gql } from '@apollo/client';

export const AUTHORIZE = gql`
  mutation authorize($email: String!, $password: String!) {
    authorize(credentials: { email: $email, password: $password }) {
      accessToken
    }
  }
`;

export const FOLLOW_COLLECTION = gql`
  mutation followCollection($collectionId: ID!) {
    followCollection(collectionId: $collectionId)
  }
`;
