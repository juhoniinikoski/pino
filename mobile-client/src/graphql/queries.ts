import { gql } from '@apollo/client';

export const GET_CHANNELS = gql`
  query Channels($first: Int, $after: String, $searchKeyword: String) {
    channels(first: $first, after: $after, searchKeyword: $searchKeyword) {
      edges {
        node {
          id
          name
          followedBy
          questions
        }
        cursor
      }
      totalCount
      pageInfo {
        endCursor
        startCursor
        hasNextPage
      }
    }
  }
`;

export const GET_FOLLOWED = gql`
  query FollowedCollections(
    $first: Int
    $after: String
    $searchKeyword: String
  ) {
    followedCollections(
      first: $first
      after: $after
      searchKeyword: $searchKeyword
      orderBy: "connectionDate"
    ) {
      edges {
        node {
          ... on Stack {
            id
            name
            followedBy
            questions
            connectionDate
            createdById
          }
          ... on Channel {
            id
            name
            followedBy
            questions
            connectionDate
          }
        }
        cursor
      }
      totalCount
      pageInfo {
        endCursor
        startCursor
        hasNextPage
      }
    }
  }
`;

export const GET_QUESTIONS = gql`
  query Questions(
    $collectionId: ID
    $createdBy: ID
    $first: Int
    $after: String
  ) {
    questions(
      collectionId: $collectionId
      createdBy: $createdBy
      first: $first
      after: $after
    ) {
      edges {
        node {
          id
          question
          createdById
          type
          answers {
            answer
            correct
            id
          }
          updatedAt
          createdAt
        }
        cursor
      }
      totalCount
      pageInfo {
        startCursor
        endCursor
        hasNextPage
        hasPreviousPage
      }
    }
  }
`;

export const GET_AUTHORIZED_USER = gql`
  query AuthorizedUser {
    authorizedUser {
      id
      email
    }
  }
`;
