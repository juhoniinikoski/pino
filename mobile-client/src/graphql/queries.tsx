import { gql } from '@apollo/client';

export const GET_CHANNELS = gql`
  query Channels($first: Int, $after: String, $searchKeyword: String) {
    channels(first: $first, after: $after, searchKeyword: $searchKeyword) {
      edges {
        node {
          id
          name
          followedBy
          questions
        }
        cursor
      }
      totalCount
      pageInfo {
        endCursor
        startCursor
        hasNextPage
      }
    }
  }
`;

export const GET_QUESTIONS = gql`
  query Questions(
    $channelId: ID
    $stackId: ID
    $createdBy: ID
    $first: Int
    $after: String
  ) {
    questions(
      channelId: $channelId
      stackId: $stackId
      createdBy: $createdBy
      first: $first
      after: $after
    ) {
      edges {
        node {
          id
          question
          createdById
          type
          answers {
            answer
            correct
            id
          }
          updatedAt
          createdAt
        }
        cursor
      }
      totalCount
      pageInfo {
        startCursor
        endCursor
        hasNextPage
        hasPreviousPage
      }
    }
  }
`;
