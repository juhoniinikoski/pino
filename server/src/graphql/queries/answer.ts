import { gql } from 'apollo-server';
import { getAnswer } from '../../services/answer/answerService';

export const typeDefs = gql`
  extend type Query {
    """
    Returns an answer.
    """
    answer(id: ID!): Answer!
  }
`;

interface Args {
  id: string | number;
}

export const resolvers = {
  Query: {
    answer: (_obj: null, { id }: Args) => getAnswer(id),
  },
};

export default {
  typeDefs,
  resolvers,
};
