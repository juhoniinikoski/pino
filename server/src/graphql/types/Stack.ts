import { gql } from 'apollo-server';
import CollectionClass from '../../models/Collection';

export const typeDefs = gql`
  type Stack {
    id: ID!
    name: String!
    questions: Int!
    followedBy: Int!
    public: Boolean!
    createdById: ID!
    createdAt: DateTime!
    updatedAt: DateTime!
    connectionDate: DateTime
  }
`;

export const resolvers = {
  Stack: {
    public: (obj: CollectionClass) => {
      return obj.public === 'true' ? true : false;
    },
  },
};

export default {
  typeDefs,
  resolvers,
};
