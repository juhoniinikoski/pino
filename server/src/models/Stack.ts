import BaseModel from './BaseModel';
import knex from '../utils/knex';
import { Id, Model } from 'objection';

class StackClass extends BaseModel {
  static idColumn = 'id';

  static tableName = 'stacks';

  id: string | number;
  name: string;
  public: string | boolean;
  followedBy: number;
  createdById: string | Id;
  createdAt: Date;
  updatedAt: Date;
  questions: number;
  connectionDate?: Date;

  static relationMappings = {
    questions: {
      relation: Model.ManyToManyRelation,
      modelClass: __dirname + '/Question',
      join: {
        from: 'stacks.id',
        through: {
          // question_stack is the join table.
          from: 'question_stack.stackId',
          to: 'question_stack.questionId',
        },
        to: 'questions.id',
      },
    },
    tags: {
      relation: Model.ManyToManyRelation,
      modelClass: __dirname + '/Channel',
      join: {
        from: 'stacks.id',
        through: {
          // stack_channel is the join table.
          from: 'stack_channel.stackId',
          to: 'stack_channel.channelId',
        },
        to: 'channels.id',
      },
    },
    followedBy: {
      relation: Model.HasManyRelation,
      modelClass: __dirname + '/UserStack',
      join: {
        from: 'stacks.id',
        to: 'user_stack.stackId',
      },
    },
  };
}

export default StackClass;

export const Stack = StackClass.bindKnex(knex);
