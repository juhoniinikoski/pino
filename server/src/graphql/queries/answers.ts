import { gql } from 'apollo-server';
import { getAnswers } from '../../services/answer/answerService';

export const typeDefs = gql`
  extend type Query {
    """
    Returns list of answers.
    """
    answers: [Answer!]
  }
`;

export const resolvers = {
  Query: {
    answers: async () => getAnswers(),
  },
};

export default {
  typeDefs,
  resolvers,
};
