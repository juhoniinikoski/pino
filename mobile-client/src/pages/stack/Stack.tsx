import { Text, StyleSheet, FlatList, View } from 'react-native';
import * as React from 'react';
import { NativeStackScreenProps } from '@react-navigation/native-stack';
import Layout from '../../components/layout/Layout';
import { LibraryStackParamList } from '../../navigation/LibraryStack';
import useQuestions from '../../hooks/useQuestions';
import { Question } from '../../utils/types';
import parseNodes from '../../utils/parseNodes';
import QuestionBox from '../../components/questionBox/QuestionBox';
import NewQuestionButton from '../../components/newQuestionButton/NewQuestionButton';
import useAuthorizedUser from '../../hooks/useAuthorizedUser';

const styles = StyleSheet.create({
  loadingText: {
    textAlign: 'center',
    textAlignVertical: 'center',
  },
  header: {
    marginTop: 16,
    marginBottom: 8,
  },
});

type Props = NativeStackScreenProps<LibraryStackParamList, 'Stack'>;

const Stack = ({ route }: Props) => {
  const { stack } = route.params;

  const { questions: raw, loading } = useQuestions(stack.id);
  const questions = raw ? parseNodes<Question>(raw) : [];

  const { user } = useAuthorizedUser();

  if (loading) {
    return (
      <Layout>
        <Text style={styles.loadingText}>Loading</Text>
      </Layout>
    );
  }

  return (
    <Layout>
      <FlatList
        testID="question-list"
        data={questions}
        keyExtractor={item => item.id}
        renderItem={({ item, index }) => (
          <QuestionBox question={item} collectionId={stack.id} index={index} />
        )}
        showsVerticalScrollIndicator={false}
        showsHorizontalScrollIndicator={false}
      />
      {user && user.id === stack.createdById && (
        <View testID="add-question">
          <NewQuestionButton stack={stack} />
        </View>
      )}
    </Layout>
  );
};

export default Stack;
