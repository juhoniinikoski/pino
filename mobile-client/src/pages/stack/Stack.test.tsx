import { MockedProvider } from '@apollo/client/testing';
import { NavigationContainer } from '@react-navigation/native';
import { render, RenderAPI, waitFor } from '@testing-library/react-native';
import * as React from 'react';
import { GET_AUTHORIZED_USER, GET_QUESTIONS } from '../../graphql/queries';
import Stack from './Stack';

/* eslint-disable no-unused-expressions */
/* eslint-disable @typescript-eslint/no-explicit-any */
/* eslint-disable no-promise-executor-return */

const mockedNavigate = jest.fn();

jest.mock('@react-navigation/native', () => {
  const actualNav = jest.requireActual('@react-navigation/native');
  return {
    ...actualNav,
    useNavigation: () => ({
      navigate: mockedNavigate,
    }),
  };
});

const mocks = [
  {
    request: {
      query: GET_QUESTIONS,
      variables: {
        stackId: 'kauppis-yh1234stack',
      },
    },
    result: {
      data: [
        {
          cursor: 'WyIyMDIyLTA1LTA3VDEyOjUwOjU4Ljk4MloiLCIxIl0=',
          node: {
            id: '1',
            question: 'Mikä seuraavista väittämistä ei pidä paikkaansa?',
            answers: [
              {
                answer:
                  'Talouspolitiikalla pyritään edistämään työllisyyttä ja talouskasvua. Yksitalouspolitiikan osa-alue on tulopolitiikka, mistä esimerkkinä on työmarkkinajärjestöjen palkkaneuvottelut.',
                correct: false,
                id: '1',
              },
              {
                answer:
                  'Laskusuhdanteessa työttömyyskorvausten ja muiden tulonsiirtojen määrä lisääntyy tyypillisesti. Nämä toimivat osaltaan Suomen valtion harjoittaman rahapolitiikan niin sanottuina automaattisina vakauttajina.',
                correct: true,
                id: '2',
              },
              {
                answer:
                  'Finanssipolitiikalla tarkoitetaan julkisen vallan tulojen ja menojen säätelyä. Valtion talousarvio on siten keskeisessä asemassa finanssipolitiikassa.',
                correct: false,
                id: '3',
              },
            ],
            updatedAt: '2022-05-07T12:50:58.982Z',
            createdAt: '2022-05-07T12:50:58.982Z',
          },
        },
        {
          cursor: 'WyIyMDIyLTA1LTA3VDExOjUwOjU4Ljk4M1oiLCIyIl0=',
          node: {
            id: '2',
            question: 'Mikä seuraavista väittämistä pitää paikkansa?',
            answers: [
              {
                answer:
                  'Rooman imperiumi ylsi laajimmillaan pohjoisessa Itämerelle ja etelässä Egyptiin. Etelässä Rooman imperiumin vaikutusalueeseen kuului lähes koko Pohjois-Afrikan rannikko.',
                correct: false,
                id: '4',
              },
              {
                answer:
                  'Kreikan valtakausi Välimeren alueen suurvaltana alkoi Rooman imperiumin luhistuttua. Kreikkalaiset laajensivat vaikutusaluettaan huomattavasti kauemmas itään, jopa Intiaan saakka.',
                correct: false,
                id: '5',
              },
              {
                answer:
                  'Jokilaaksojen korkeakulttuureissa kehittyi monia nykyisinkin käytössä olevia keksintöjä. Näitä olivat esimerkiksi pyörä ja aura. Myös nykyisen kaltainen ajanlasku kehittyi silloin.',
                correct: true,
                id: '6',
              },
            ],
            updatedAt: '2022-05-07T11:50:58.983Z',
            createdAt: '2022-05-07T11:50:58.983Z',
          },
        },
        {
          cursor: 'WyIyMDIyLTA1LTA3VDEwOjUwOjU4Ljk4M1oiLCIzIl0=',
          node: {
            id: '3',
            question: 'Mikä seuraavista väittämistä pitää paikkansa?',
            answers: [
              {
                answer:
                  'Eri maiden tuotantoon liittyvä työnjako hämärtyi globalisaation myötä 1800-luvulla.',
                correct: false,
                id: '7',
              },
              {
                answer:
                  '1800-luvun lopun tavarakauppaa vauhditti kuljetuskustannusten aleneminen ja kuljetusmahdollisuuksien monipuolistuminen.',
                correct: true,
                id: '8',
              },
              {
                answer:
                  'Globalisaatio ilmenee nykyisin edelleen saman toimialan yritysten maailmanlaajuisena kilpailuna, missä työntekijät jäävät pääasiassa kansainvälisen keskinäisen kilpailun ulkopuolelle.',
                correct: false,
                id: '9',
              },
            ],
            updatedAt: '2022-05-07T10:50:58.983Z',
            createdAt: '2022-05-07T10:50:58.983Z',
          },
        },
        {
          cursor: 'WyIyMDIyLTA1LTA3VDEwOjUwOjU4Ljk4M1oiLCI0Il0=',
          node: {
            id: '4',
            question:
              'Verohallinnon vuonna 2020 keräämien verojen nettokertymät olivat yhteensä 69 217 miljoonaa euroa, joka oli 1 141 miljoonaa euroa (-1,6 %) vähemmän kuin vuonna 2019, mutta enemmän kuin sitä edeltävinä vuosina. Vuonna 2020 kerätyistä veroista 59,5 prosenttia tilitettiin valtiolle ja 34,6 prosenttia kunnille. Muita veronsaajia ovat lisäksi seurakunnat 1,3 prosentin ja Kela 4,6 prosentin osuuksilla. (Lähde: vero.fi)\n\nMikä seuraavista väittämistä pitää paikkaansa?',
            answers: [
              {
                answer:
                  'Kiinteistöveroa peritään, kun kiinteistö vaihtaa omistajaa. Kiinteistövero tilitetään kiinteistön sijaintikunnalle.',
                correct: false,
                id: '10',
              },
              {
                answer:
                  'Henkilöiden tuloverotus on verokertymän kannalta merkittävä. Se tilitetään valtiolle, kunnille ja seurakunnille.',
                correct: true,
                id: '11',
              },
              {
                answer:
                  'Esimerkiksi asunto- ja kiinteistökauppojen yhteydessä perittävä varallisuusvero muodostaa osan valtiolle tilitettävistä veroista.',
                correct: false,
                id: '12',
              },
            ],
            updatedAt: '2022-05-07T10:50:58.983Z',
            createdAt: '2022-05-07T10:50:58.983Z',
          },
        },
      ],
    },
  },
  {
    request: {
      query: GET_AUTHORIZED_USER,
    },
    result: {
      data: {
        authorizedUser: {
          id: 'bbe42984-051b-4a01-b45d-b8d29c32200c',
          email: 'testi1@gmail.com',
        },
      },
    },
  },
];

const createTestProps = (props: Record<string, unknown>) => ({
  navigation: {
    navigate: jest.fn(),
  },
  route: {
    params: {
      stack: {
        followedBy: 2,
        id: 'kauppis-yh1234stack',
        name: 'kauppis-yh',
        createdById: 'bbe42984-051b-4a01-b45d-b8d29c32200c',
        questions: 4,
      },
    },
  },
  ...props,
});

const createTestProps2 = (props: Record<string, unknown>) => ({
  navigation: {
    navigate: jest.fn(),
  },
  route: {
    params: {
      stack: {
        followedBy: 2,
        id: 'yomatematiikka1234stack',
        name: 'yomatematiikka',
        createdById: 'jokumuu',
        questions: 4,
      },
    },
  },
  ...props,
});

describe('render tests', () => {
  let component: RenderAPI;
  let props: any;

  beforeEach(() => {
    props = createTestProps({});
    component = render(
      <MockedProvider mocks={mocks} addTypename={false}>
        <NavigationContainer>
          <Stack route={props.route} navigation={props.navigation} />
        </NavigationContainer>
      </MockedProvider>,
    );
  });

  it('should display a name of the stack', async () => {
    await waitFor(async () => {
      await new Promise(resolve => setTimeout(resolve, 0));
      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  it('should show loading indicator when questions are loaded', async () => {
    await waitFor(async () => {
      expect(component.getByText('Loading')).toBeTruthy;
      await new Promise(resolve => setTimeout(resolve, 0));
      expect(component).toMatchSnapshot();
    });
  });

  it('should render a list containing questions succesfully', async () => {
    await waitFor(async () => {
      await new Promise(resolve => setTimeout(resolve, 0));
      expect(component.getAllByTestId('question-list').length).toBe(1);
      expect(component).toMatchSnapshot();
    });
  });

  it('should render add question button succesfully', async () => {
    await waitFor(async () => {
      await new Promise(resolve => setTimeout(resolve, 0));
      expect(component.getByTestId('add-question')).toBeDefined;
      expect(component).toMatchSnapshot();
    });
  });
});

describe('add question button responsibility', () => {
  let component: RenderAPI;
  let props: any;

  beforeEach(() => {
    props = createTestProps2({});
    component = render(
      <MockedProvider mocks={mocks} addTypename={false}>
        <NavigationContainer>
          <Stack route={props.route} navigation={props.navigation} />
        </NavigationContainer>
      </MockedProvider>,
    );
  });

  it('if stack is not made by authorized user, it should not display new question button', async () => {
    await waitFor(async () => {
      await new Promise(resolve => setTimeout(resolve, 0));
      expect(component.queryByTestId('add-question')).toBeNull();
      expect(component).toMatchSnapshot();
    });
  });
});
