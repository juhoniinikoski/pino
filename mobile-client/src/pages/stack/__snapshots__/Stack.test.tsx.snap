// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`add question button responsibility if stack is not made by authorized user, it should not display new question button 1`] = `
<RNCSafeAreaView
  edges={
    Array [
      "left",
      "right",
    ]
  }
  style={
    Array [
      Object {
        "display": "flex",
        "flex": 1,
        "marginHorizontal": 16,
      },
      undefined,
    ]
  }
>
  <RCTScrollView
    data={Array []}
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    showsHorizontalScrollIndicator={false}
    showsVerticalScrollIndicator={false}
    stickyHeaderIndices={Array []}
    testID="question-list"
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View />
  </RCTScrollView>
</RNCSafeAreaView>
`;

exports[`render tests should display a name of the stack 1`] = `
<RNCSafeAreaView
  edges={
    Array [
      "left",
      "right",
    ]
  }
  style={
    Array [
      Object {
        "display": "flex",
        "flex": 1,
        "marginHorizontal": 16,
      },
      undefined,
    ]
  }
>
  <RCTScrollView
    data={Array []}
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    showsHorizontalScrollIndicator={false}
    showsVerticalScrollIndicator={false}
    stickyHeaderIndices={Array []}
    testID="question-list"
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View />
  </RCTScrollView>
  <View
    testID="add-question"
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#EFEFEF",
          "borderRadius": 100,
          "bottom": 16,
          "display": "flex",
          "height": 64,
          "justifyContent": "center",
          "position": "absolute",
          "right": 0,
          "shadowColor": "black",
          "shadowOffset": Object {
            "height": 4,
            "width": 0,
          },
          "shadowOpacity": 0.08,
          "shadowRadius": 8,
          "width": 64,
        }
      }
      testID="button"
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "black",
              "fontSize": 28,
            },
            Object {
              "marginLeft": 6,
            },
            Object {
              "fontFamily": "FontAwesome",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
  </View>
</RNCSafeAreaView>
`;

exports[`render tests should render a list containing questions succesfully 1`] = `
<RNCSafeAreaView
  edges={
    Array [
      "left",
      "right",
    ]
  }
  style={
    Array [
      Object {
        "display": "flex",
        "flex": 1,
        "marginHorizontal": 16,
      },
      undefined,
    ]
  }
>
  <RCTScrollView
    data={Array []}
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    showsHorizontalScrollIndicator={false}
    showsVerticalScrollIndicator={false}
    stickyHeaderIndices={Array []}
    testID="question-list"
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View />
  </RCTScrollView>
  <View
    testID="add-question"
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#EFEFEF",
          "borderRadius": 100,
          "bottom": 16,
          "display": "flex",
          "height": 64,
          "justifyContent": "center",
          "position": "absolute",
          "right": 0,
          "shadowColor": "black",
          "shadowOffset": Object {
            "height": 4,
            "width": 0,
          },
          "shadowOpacity": 0.08,
          "shadowRadius": 8,
          "width": 64,
        }
      }
      testID="button"
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "black",
              "fontSize": 28,
            },
            Object {
              "marginLeft": 6,
            },
            Object {
              "fontFamily": "FontAwesome",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
  </View>
</RNCSafeAreaView>
`;

exports[`render tests should render add question button succesfully 1`] = `
<RNCSafeAreaView
  edges={
    Array [
      "left",
      "right",
    ]
  }
  style={
    Array [
      Object {
        "display": "flex",
        "flex": 1,
        "marginHorizontal": 16,
      },
      undefined,
    ]
  }
>
  <RCTScrollView
    data={Array []}
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    showsHorizontalScrollIndicator={false}
    showsVerticalScrollIndicator={false}
    stickyHeaderIndices={Array []}
    testID="question-list"
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View />
  </RCTScrollView>
  <View
    testID="add-question"
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#EFEFEF",
          "borderRadius": 100,
          "bottom": 16,
          "display": "flex",
          "height": 64,
          "justifyContent": "center",
          "position": "absolute",
          "right": 0,
          "shadowColor": "black",
          "shadowOffset": Object {
            "height": 4,
            "width": 0,
          },
          "shadowOpacity": 0.08,
          "shadowRadius": 8,
          "width": 64,
        }
      }
      testID="button"
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "black",
              "fontSize": 28,
            },
            Object {
              "marginLeft": 6,
            },
            Object {
              "fontFamily": "FontAwesome",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
  </View>
</RNCSafeAreaView>
`;

exports[`render tests should show loading indicator when questions are loaded 1`] = `
<RNCSafeAreaView
  edges={
    Array [
      "left",
      "right",
    ]
  }
  style={
    Array [
      Object {
        "display": "flex",
        "flex": 1,
        "marginHorizontal": 16,
      },
      undefined,
    ]
  }
>
  <RCTScrollView
    data={Array []}
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    showsHorizontalScrollIndicator={false}
    showsVerticalScrollIndicator={false}
    stickyHeaderIndices={Array []}
    testID="question-list"
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View />
  </RCTScrollView>
  <View
    testID="add-question"
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#EFEFEF",
          "borderRadius": 100,
          "bottom": 16,
          "display": "flex",
          "height": 64,
          "justifyContent": "center",
          "position": "absolute",
          "right": 0,
          "shadowColor": "black",
          "shadowOffset": Object {
            "height": 4,
            "width": 0,
          },
          "shadowOpacity": 0.08,
          "shadowRadius": 8,
          "width": 64,
        }
      }
      testID="button"
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "black",
              "fontSize": 28,
            },
            Object {
              "marginLeft": 6,
            },
            Object {
              "fontFamily": "FontAwesome",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
  </View>
</RNCSafeAreaView>
`;
